npm/get_binary.js

```diff
@@ -0,0 +1,30 @@
+const os = require("os");
+const { Binary } = require("./binary_install");
+
+function getPlatform() {
+  const operatingSystem = os.type();
+  const architecture = os.arch();
+
+  const NODEJS_OS_TO_GOOS = {
+    Linux: "linux",
+    Darwin: "darwin",
+    Windows_NT: "windows",
+  };
+  const NODEJS_ARCH_TO_GOARCH = { x64: "amd64", arm64: "arm64", ia32: "386" };
+
+  return {
+    platform: NODEJS_OS_TO_GOOS[operatingSystem],
+    arch: NODEJS_ARCH_TO_GOARCH[architecture],
+  };
+}
+
+function getBinary() {
+  const { platform, arch } = getPlatform();
```

Comment:
Can we throw error if platform not supported?

Reply:
Umm yes sounds good